File: Python/238. Product of Array Except Self.py
class Solution:
    def productExceptSelf(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        ans = list()
        res, cnt = 1, 0
        for num in nums:
            if num == 0:
                cnt += 1
            else:
                res *= num
        for i in range(len(nums)):
            if cnt >= 2:
                ans.append(0)
            elif cnt == 1:
                if nums[i] == 0:
                    ans.append(res)
                else:
                    ans.append(0)
            else:
                ans.append(res // nums[i])
        return ans

File: C++/238. Product of Array Except Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans(nums.size(), 0);
        int sum = 1, cnt = 0;
        for(int num: nums)
        {
            if(num) sum *= num;
            else
                cnt++;
        }
        for(int i = 0;i<nums.size(); i++)
        {
            if(cnt >= 2)
                continue;
            else if(cnt == 1)
            {
                if(nums[i] == 0)
                    ans[i] = sum;
            }
            else
                ans[i] = sum / nums[i];
        }
        return ans;
    }
};